torch/_jit_internal.py

```diff
@@ -114,9 +114,11 @@ def baz():
     frame = inspect.currentframe()
     i = 0
     while i < frames_up + 1:
+        assert frame is not None
```

Comment:
Does this not work if you put the assert on the line right below `inspect.currentframe()`? (it wouldn't surprise me if not, `mypy` is fragile with this kind of type narrowing)

Technically `assert`'s are used for asserting things that _must_ be true. In this case that is the case (because the reason for `None` is the below), so the change is correct - just unfortunate the same `assert` must be used twice and inside a for-loop.

_CPython implementation detail: This function relies on Python stack frame support in the interpreter, which isn’t guaranteed to exist in all implementations of Python. If running in an implementation without Python stack frame support this function returns None._

Reply:
Yeah, unfortunately it doesn't work. If I move the assertion to the line below `current_frame = inspect.currentframe()`, mypy raises an error.